assets/sass/07-utilities/color-palette.scss

```diff
@@ -43,7 +43,7 @@
 // Gutenberg background-color options
 .has-background {
 
-	&:not(.has-background-background-color) a:not(.wp-block-button__link),
```

Comment:
Removing the `:not(.wp-block-button__link)` is intentional too — this seems to be unnecessary, since the button link styles have a higher specificity.